src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -9,6 +9,25 @@ public class RegistrationServiceImpl implements RegistrationService {
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RuntimeException("User without LOGIN can not be registered!");
+        }
+        if (user.getAge() == null) {
+            throw new RuntimeException("User without AGE can not be registered!");
+        }
+        if (user.getPassword() == null) {
+            throw new RuntimeException("User without PASSWORD can not be registered!");
+        }
+        if (storageDao.get(user.getLogin()) != null) {
+            throw new RuntimeException("User with such login already exists!");
+        }
+        if (user.getAge() < 18) {
+            throw new RuntimeException("User is to be at least 18 years!");
+        }
+        if (user.getPassword().length() < 6) {
+            throw new RuntimeException("User`s password is to be at least 6 characters!");
```

Comment:
lets make numbers 6 and 18 as constant fields

Reply:
moved to Constants